### PR TITLE
Allow to configure domain to create routes

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
Allow to configure domain for routes.
The host will be constructed as `<route-name>-<namespace>.<domain>`

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-1466

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [Custom resource definition file is up to date](https://github.com/eclipse/che-operator/blob/master/README.md#updating-custom-resource-definition-file)
- [x] [Nightly OLM bundle is up to date](https://github.com/eclipse/che-operator/blob/master/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: pkg/deploy/devfile-registry/devfile_registry.go [hunk 1]
```diff
@@ -34,8 +34,13 @@ type DevFileRegistryConfigMap struct {
...
 	devfileRegistryURL := deployContext.CheCluster.Spec.Server.DevfileRegistryUrl
 	if !deployContext.CheCluster.Spec.Server.ExternalDevfileRegistry {
-		additionalLabels := (map[bool]string{true: deployContext.CheCluster.Spec.Server.DevfileRegistryRoute.Labels, false: deployContext.CheCluster.Spec.Server.DevfileRegistryIngress.Labels})[util.IsOpenShift]
-		endpoint, done, err := expose.Expose(deployContext, cheHost, deploy.DevfileRegistryName, additionalLabels, deploy.DevfileRegistryName)
+		endpoint, done, err := expose.Expose(
+			deployContext,
+			cheHost,
+			deploy.DevfileRegistryName,
+			deployContext.CheCluster.Spec.Server.PluginRegistryRoute,
+			deployContext.CheCluster.Spec.Server.PluginRegistryIngress,
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/devfile-registry/devfile_registry.go:42]: Sorry, why do you use PluginRegistryRoute and PluginRegistryIngress for "devfile" registry expose?

Comment (tolusha) [reply pkg/deploy/devfile-registry/devfile_registry.go:42]: that was copy/paste ^(

File: pkg/deploy/route_test.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+						"app.kubernetes.io/component":  "test-component",
+						"app.kubernetes.io/instance":   "che",
+						"app.kubernetes.io/managed-by": "che-operator",
+						"app.kubernetes.io/name":       "che",
+					},
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion:         "org.eclipse.che/v1",
+							Kind:               "CheCluster",
+							Controller:         &_true,
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/route_test.go:76]: You can use util.NewBoolPointer(true)

File: pkg/deploy/route_test.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+							BlockOwnerDeletion: &_true,
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/route_test.go:77]: You can use util.NewBoolPointer(true)

File: pkg/deploy/ingres_test.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+						"app.kubernetes.io/component":  "test-component",
+						"app.kubernetes.io/instance":   "che",
+						"app.kubernetes.io/managed-by": "che-operator",
+						"app.kubernetes.io/name":       "che",
+					},
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion:         "org.eclipse.che/v1",
+							Kind:               "CheCluster",
+							Controller:         &_true,
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/ingres_test.go:75]: You can use util.NewBoolPointer(true)

File: pkg/deploy/ingres_test.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+							BlockOwnerDeletion: &_true,
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/ingres_test.go:76]: You can use util.NewBoolPointer(true)

File: pkg/deploy/route_test.go [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+						"app.kubernetes.io/component":  "test-component",
+						"app.kubernetes.io/instance":   "che",
+						"app.kubernetes.io/managed-by": "che-operator",
+						"app.kubernetes.io/name":       "che",
+					},
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion:         "org.eclipse.che/v1",
+							Kind:               "CheCluster",
+							Controller:         &_true,
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/route_test.go:127]: You can use util.NewBoolPointer(true)

File: pkg/deploy/route_test.go [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+							BlockOwnerDeletion: &_true,
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/route_test.go:128]: You can use util.NewBoolPointer(true)

File: pkg/deploy/server/che_configmap_test.go [hunk 8]
```diff
@@ -153,12 +153,14 @@ func TestConfigMap(t *testing.T) {
...
 			cli := fake.NewFakeClientWithScheme(scheme.Scheme, testCase.initObjects...)
+			nonCachedClient := fake.NewFakeClientWithScheme(scheme.Scheme, testCase.initObjects...)
 
 			deployContext := &deploy.DeployContext{
 				CheCluster: testCase.cheCluster,
 				ClusterAPI: deploy.ClusterAPI{
-					Client: cli,
-					Scheme: scheme.Scheme,
+					Client:          cli,
+					NonCachedClient: nonCachedClient,
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/server/che_configmap_test.go:162]: You can simplify test a bit:

```
NonCachedClient: cli
```

But up to you.

**Review by AndrienkoAleksandr — APPROVED** [review]

**Pull request merged**